src/coord/src/coord.rs

```diff
@@ -277,134 +278,7 @@ where
-                .await;
-
-            for (schema_name, schema_id, schema_oid, items) in schemas {
-                self.report_schema_update(
-                    schema_id,
-                    schema_oid,
-                    Some(database_id),
-                    &schema_name,
-                    1,
-                )
-                .await;
-
-                for (item_name, item_id) in items {
-                    if let Some(oid) = tables_to_report.remove(&item_id) {
-                        self.report_table_update(item_id, oid, schema_id, &item_name, 1)
-                            .await;
-                    } else if let Some(oid) = sources_to_report.remove(&item_id) {
-                        self.report_source_update(item_id, oid, schema_id, &item_name, 1)
-                            .await;
-                    } else if let Some(oid) = views_to_report.remove(&item_id) {
-                        self.report_view_update(item_id, oid, schema_id, &item_name, 1)
-                            .await;
-                    } else if let Some(oid) = sinks_to_report.remove(&item_id) {
-                        self.report_sink_update(item_id, oid, schema_id, &item_name, 1)
-                            .await;
-                    }
-                }
-            }
-        }
-        let ambient_schemas: Vec<(String, i64, u32, Vec<(String, GlobalId)>)> = self
-            .catalog
-            .ambient_schemas()
-            .map(|(schema_name, schema)| {
-                (
-                    schema_name.to_string(),
-                    schema.id,
-                    schema.oid,
-                    schema
-                        .items
-                        .iter()
-                        .map(|(name, id)| (name.clone(), *id))
-                        .collect(),
-                )
-            })
-            .collect();
-        for (schema_name, schema_id, schema_oid, items) in ambient_schemas {
-            self.report_schema_update(schema_id, schema_oid, None, &schema_name, 1)
-                .await;
-
-            for (item_name, item_id) in items {
-                if let Some(oid) = tables_to_report.remove(&item_id) {
-                    self.report_table_update(item_id, oid, schema_id, &item_name, 1)
-                        .await;
-                } else if let Some(oid) = sources_to_report.remove(&item_id) {
-                    self.report_source_update(item_id, oid, schema_id, &item_name, 1)
-                        .await;
-                } else if let Some(oid) = views_to_report.remove(&item_id) {
-                    self.report_view_update(item_id, oid, schema_id, &item_name, 1)
-                        .await;
-                } else if let Some(oid) = sinks_to_report.remove(&item_id) {
-                    self.report_sink_update(item_id, oid, schema_id, &item_name, 1)
-                        .await;
-                }
-            }
-        }
+        self.process_catalog_events(events).await?;
```

Comment:
🙏

Reply:
I've got a follow up PR (hopefully) that gets rid of this bootstrap method entirely! 🤞